.github/workflows/rust.yml

```diff
@@ -0,0 +1,22 @@
+name: Rust
```

Comment:
A better commit title would be:

workflow: Add a rust-based CI workflow for the project